src/components/refactor/dialogs/delete-attaching-line/delete-attaching-line-dialog.js

```diff
@@ -0,0 +1,138 @@
+    [LINE_TO_ATTACH_TO_1_ID]: null,
+    [LINE_TO_ATTACH_TO_2_ID]: null,
+    [REPLACING_LINE_ID]: '',
+    [REPLACING_LINE_NAME]: '',
+};
+
+const schema = yup
+    .object()
+    .shape({
+        [ATTACHED_LINE_ID]: yup.string().nullable().required(),
+        [LINE_TO_ATTACH_TO_1_ID]: yup.string().nullable().required(),
+        [LINE_TO_ATTACH_TO_2_ID]: yup.string().nullable().required(),
+        [REPLACING_LINE_ID]: yup.string().required(),
+        [REPLACING_LINE_NAME]: yup.string(),
+    })
+    .required();
+
+/**
+ * Dialog to delete attaching line.
+ * @param studyUuid the study we are currently working on
+ * @param currentNode the node we are currently working on
+ * @param editData the data to edit
+ * @param dialogProps props that are forwarded to the generic ModificationDialog component
+ */
+const DeleteAttachingLineDialog = ({
+    studyUuid,
+    currentNode,
+    editData,
+    ...dialogProps
+}) => {
+    const currentNodeUuid = currentNode?.id;
+
+    const { snackError } = useSnackMessage();
+
+    const methods = useForm({
+        defaultValues: emptyFormData,
+        resolver: yupResolver(schema),
+    });
+
+    const { reset } = methods;
+
+    const fromEditDataToFormValues = useCallback(
+        (editData) => {
+            reset({
+                [ATTACHED_LINE_ID]: editData.attachedLineId,
+                [LINE_TO_ATTACH_TO_1_ID]: editData.lineToAttachTo1Id,
+                [LINE_TO_ATTACH_TO_2_ID]: editData.lineToAttachTo2Id,
+                [REPLACING_LINE_ID]: editData.replacingLine1Id,
+                [REPLACING_LINE_NAME]: editData.replacingLine1Name,
+            });
+        },
+        [reset]
+    );
+
+    useEffect(() => {
+        if (editData) {
+            fromEditDataToFormValues(editData);
+        }
+    }, [fromEditDataToFormValues, editData]);
+
+    const onSubmit = useCallback(
+        (formData) => {
+            deleteAttachingLine(
+                studyUuid,
+                currentNodeUuid,
+                editData ? editData.uuid : undefined,
+                formData[LINE_TO_ATTACH_TO_1_ID],
+                formData[LINE_TO_ATTACH_TO_2_ID],
+                formData[ATTACHED_LINE_ID],
+                formData[REPLACING_LINE_ID],
+                sanitizeString(formData[REPLACING_LINE_NAME])
+            ).catch((error) => {
+                snackError({
+                    messageTxt: error.message,
+                    headerId: 'DeleteAttachingLineError',
+                });
+            });
+        },
+        [currentNodeUuid, editData, snackError, studyUuid]
+    );
+
+    const clear = useCallback(() => {
+        reset(emptyFormData);
+    }, [reset]);
+
+    return (
+        <FormProvider validationSchema={schema} {...methods}>
+            <ModificationDialog
+                fullWidth
+                maxWidth="md"
+                onClear={clear}
+                onSave={onSubmit}
+                aria-labelledby="dialog-delete-voltage-level-on-line"
```

Comment:
I see it was here before, but it looks wrong to me

Reply:
yes bad name